### PR TITLE
[C-5557] adds audiences support

### PR DESCRIPTION
**Pull request by suhasdeshpande**
## Description of the change

Implements `/audiences` API in Ruby SDK

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Review by cgradwohl — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -455,6 +455,57 @@ Any other errors from the API are thrown as a CourierAPIError. Catch these error
...
+        "body" => "Welcome to Courier!"
+      },
+      "data" => {
+        "name" => "Ruby"
+      }
+    }
+  })
+  puts res.code # the HTTP response code
+  puts res.request_id # if the code is 202, this will be the Courier request ID for this message
+rescue Courier::CourierAPIError => re #error sent from from the API
```
Comment (abdul-courier) [inline README.md:504]:
```suggestion
rescue Courier::CourierAPIError => re #error sent from the API
```

File: README.md [hunk 2]
```diff
@@ -468,3 +519,7 @@ Bug reports and pull requests are welcome on GitHub at https://github.com/trycou
 ## License
 
 The gem is available as open source under the terms of the [MIT License](https://opensource.org/licenses/MIT).
+
+```
+
+```
```
Comment (tk26) [inline README.md:525]: whats this for?

**Review by abdul-courier — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (suhasdeshpande) [inline README.md:525]:
```suggestion
```

Comment (suhasdeshpande) [reply README.md]: probably typo

**Review by tk26 — DISMISSED** [review]

**Review by tk26 — APPROVED** [review]

**Pull request merged**